### ISSUE TITLE
RxJava 2.0

### ISSUE THREAD

**Issue by jesperldk** [intendia-oss/rxjava-gwt#4]
Hi
I'm just started with RxJava on the server side and GWT on the client side.
RxJava GWT looks like just what I need next. However, I really would like to use RxJava 2.0. Any thoughts/progress on that?
Thanks!

**Comment by ibaca** [comment]
Last time I'll get asked for RxJava 2.0 😉:
> I stop updating rxjava-gwt because currently each rx release requires too much manual work and there are no easy test to check everything works correctly. AND! current state is more than enough to start using RxJava in a production GWT app. I think that currently it is more important to develop some tools to make RxJava useful in the client side and this is why I have been spending much more time in https://github.com/intendia-oss/autorest, https://github.com/intendia-oss/rxgwt and https://github.com/intendia-oss/autorpc-gwt. And examples of how to use RxJava in the client side like https://github.com/ibaca/autorest-nominatim-example, https://github.com/ibaca/rxcanvas-gwt or https://github.com/ibaca/rxsnake-gwt.
>
>All this tools and examples are intended to increase the numbers of users using rxjava-gwt and this will make possible to convince the rxjava team to integrate the GWT support in the main project which is the final goal. But, as you can see, there are not enough activity and projects using rxjava-gwt to  convince the rxjava team to add support to GWT. So if you start using rxjava in the client side, try to make some noise ;), add examples, issues, tweets, articles… etc.
>
>RxJava 2 requires the same effort than RxJava 1.x, it is quite probably that the next rxjava-gwt release will be a RxJava 2 and not the last version of RxJava 1.x. Another thing that is slowing down release cycle is that RxJava uses gradle, and I'm not much experienced compiling GWT using gradle, but I definitely will start using gradle for compiling GWT in 2017, so this might help to start releasing new versions of RxJava GWT.

So, hehe please use it and make sound. We use it extensively internally, both in the server side and the client side. I really want to start using RxJava 2, but for us is still not critical. If the project go up to 100 stars I'll definitely include RxJava 2 😛!

**Comment by jesperldk** [comment]
Cool, you got at star for starters ;-)

**Comment by ibaca** [comment]
I'm going to leave it opened, hehe to make back-pressure...

**Comment by lyubomyr-shaydariv** [comment]
> If the project go up to 100 stars I'll definitely include RxJava 2 :stuck_out_tongue:!

★★★★★ ★★★★★ ★★★★★ ★★★★★
★★★★★ ★★★★★ ★★★★★ ★★★★★
★★★★★ ★★★★★ ★★★★★ ★★★★★
★★★★★ ★★★★★ ★★★★★ ★★★★★
★★★★★ ★★★★★ ★★★★★ ★★★★★

:D Anyway, it would be really awesome to have RxJava 2 support. I hope to have it soon or someday. And thanks for your great work!

**Comment by ibaca** [comment]
RxJava 2.x is here! hehe, but there is an obscure GWT compilation problem that makes it super-slow to compile! I hope I'll fix it in the next days... anyway if you are quite patient you can try it out using the snapshot version from sonatype:
https://oss.sonatype.org/content/repositories/snapshots/com/intendia/gwt/rxjava-gwt/2.x-SNAPSHOT/
If travis can compile it (https://travis-ci.org/intendia-oss/rxjava-gwt/builds/263608529), you can do it too 😜!

As soon as this problem get fixed I'll make a release to central.

**Comment by lyubomyr-shaydariv** [comment]
@ibaca Wow! :) I'm going to check it soon. Thank you very and very much!

**Comment by jesperldk** [comment]
:-D
Any new features, or is it mostly a straight move to RxJava 2.0?

**Comment by ibaca** [comment]
I have updated the README 😛, this is the new feature hehe… no, no new features, this lib is the minimal work to make RxJava GWT compatible, GWT specific features are added to https://github.com/intendia-oss/rxgwt. And bc of the JDT compiler bug, now each time you compile anything with RxJava 2 it requires +3min of compilation, even dummy project previously compiled in 20seconds, so... updating rxgwt is super-annoying until it gets solved.

The eclipse JDT compiler problem seems pretty ugly, I really hope I can fix or force to fix it ASAP, one good thing is that looks like this is affecting all compilations but it is critical in RxJava bc I think that it produces an exponential slowdown relative to the number of generics types. So hopefully when this problem gets solves all GWT compilation will get a small improvement on compilation times.

**Comment by ibaca** [comment]
Blocked by https://bugs.eclipse.org/bugs/show_bug.cgi?id=521438, this bug makes GWT take >4min to compile plain RxJava 2.x!

**Comment by ibaca** [comment]
The bug get fixed some week ago, and now the patch should be released by eclipse JDT and later on integrated in GWT, 😥 this may take a while so, in the mean time I have patched the last GWT 2.8.1 including the fix. I have deployed in this repo and can be used just adding a mvn repo and using GWT version `2.8.1-rx1`.
```
<repository>
    <id>rxjava-gwt-repo</id>
    <url>https://raw.githubusercontent.com/intendia-oss/rxjava-gwt/mvn-repo/</url>
</repository>
```
I'm going to make a rxjava 2.x release now.
